File: server/src/main/java/ru/trett/rss/models/Feed.java
package ru.trett.rss.models;

import com.fasterxml.jackson.annotation.JsonFormat;

import jakarta.validation.constraints.NotEmpty;

import java.time.LocalDateTime;

public class Feed {

    public long id;
    @NotEmpty public String title;
    @NotEmpty public String guid;
    @NotEmpty public String channelTitle;
    public String description;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm")
    public LocalDateTime pubDate;

    public String link;
    public boolean read;
}

File: server2/src/main/scala/ru/trett/rss/server/models/Feed.scala
package ru.trett.rss.server.models

import java.time.OffsetDateTime

case class Feed(
    link: String,
    channelId: Long,
    title: String,
    description: String,
    pubDate: Option[OffsetDateTime] = None,
    isRead: Boolean = false
)
